HowToFIDO.md

```diff
@@ -270,26 +273,26 @@ Reauthentication might happen for the following reasons:
     re-confirm control over the user session.
 
 Let’s look at the last case first: when it’s time to re-authenticate for
-a sensitive action, check whether you have a credential id for this user
-*for the purpose of reauthentication*, i.e., the kind of credential id
-obtained from [<span class="underline">opting the user into FIDO-based
-reauthentication</span>](#opting-into-fido-based-reauthentication). Make
-sure it’s associated with the user *and* device - for example, check a
-cookie or read from local storage.
-
-If *no credential id is available*, serve a traditional login challenge
+a sensitive action, check whether you have a `2FACapableFIDOCredsAvailable` flag for this user
+*for the purpose of reauthentication*.
+
+If *no `2FACapableFIDOCredsAvailable` flag is available*, serve a traditional login challenge
 suitable for reauthentication\[8\], for example:
 
 ![password reauthentication](images/image15.png)
 
-If, however, you *do* discover a credential id for the current session,
-then you can use FIDO-based reauthentication:
+If, however, you *do* have `2FACapableFIDOCredsAvailable` flag for the current session,
+then you can use FIDO-based reauthentication.
+Fetch all `2FACapableFidoCredential` credentials from the server for this user.
+Don't fetch non `2FACapableFidoCredential` credentials like U2F credentials.
+Prefer using long lived local session key while fetching `2FACapableFidoCredential` credentials from the server.
```

Comment:
I think this needs to be stronger that `Prefer`. I don't think it should be permitted to fetch credentials from the server for a user without some long-lived session key as otherwise there is unauthenticated credential leakage.